Content.Server/Atmos/Monitor/Systems/AirAlarmSystem.cs

```diff
@@ -468,11 +468,16 @@ private string GetPort(AirAlarmComponent comp)
     /// <param name="uiOnly">Whether this change is for the UI only, or if it changes the air alarm's operating mode. Defaults to true.</param>
     public void SetMode(EntityUid uid, string origin, AirAlarmMode mode, bool uiOnly = true, AirAlarmComponent? controller = null)
     {
-        if (!Resolve(uid, ref controller) || controller.CurrentMode == mode)
+        if (!Resolve(uid, ref controller))
         {
             return;
         }
 
+        if (controller.PanicWireCut)
+        {
+            mode = AirAlarmMode.Panic;
+        }
+
         controller.CurrentMode = mode;
```

Comment:
```suggestion
        if (controller.CurrentMode == mode)
            return;

        controller.CurrentMode = mode;
```
We keep the check here to skip the rest of the code if nothing changed.